### PR TITLE
Add sink methods get() set() has()

### PR DESCRIPTION
**Pull request by sveisvei**
## Status
**READY**

## Description
Switch to use jest and FINN.no eslint config.

Adds async methods to support setting and gettings files without streams, needed for simpler meta-data storage interface. 

## Todos
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,6 @@ language: node_js
 
 node_js:
   - "8"
```
Comment (SimenB) [inline .travis.yml]: should add `9` as well

**Comment by SimenB** [comment]
I'd like to see "lint and jest" in its own PR, so that it's easier to review the `get`, `set`, `has` in isolation

File: test/sink.test.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    const sink = new Sink();
+
+    expect(await sink.set('some-key-1', 'value-1')).toBe(true);
+    expect(await sink.has('some-key-1')).toBe(true);
+});
+
+test('.writer() - no value for "type" argument - should throw', () => {
+    const sink = new Sink();
+    expect(() => {
+        const dest = sink.writer(); // eslint-disable-line
```
Comment (digitalsadhu) [inline test/sink.test.js]: Do you need to assign the result of `sink.writer()` to a variable?

**Comment by SimenB** [comment]
Yes please 🙂

File: test/sink.test.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        expect(file).toBeDefined();
+        done();
+    });
+    source.pipe(dest);
+});
+
+test('.reader() - no value for "file" argument - should throw', () => {
+    const sink = new Sink();
+    expect(() => {
+        const source = sink.reader(); // eslint-disable-line
```
Comment (digitalsadhu) [inline test/sink.test.js]: here also?

**Comment by digitalsadhu** [comment]
Updated project meta files, just need to enable semantic-release. (and use semantic commit messages)

**Comment by SimenB** [comment]
Can the meta stuff be in a separate PR?

**Comment by digitalsadhu** [comment]
Such work, much doing, very typing. ok WoW.

**Comment by sveisvei** [comment]
@digitalsadhu Sure thing, make them boring indeed.

File: test/sink.test.js (hunk 2 above)
Comment (sveisvei) [reply test/sink.test.js]: will fix

File: test/sink.test.js (hunk 3 above)
Comment (sveisvei) [reply test/sink.test.js]: will fix

**Pull request closed without merging**